src/mmds/src/data_store.rs

```diff
@@ -107,40 +115,40 @@ impl Mmds {
         };
 
         match value {
-            Some(val) => {
+            Some(val) => match &self.response_format {
+                ResponseFormat::Imds => Mmds::get_imds_format_response(val),
+                ResponseFormat::Json => Ok(val
+                    .as_str()
+                    .map_or(val.to_string(), |val_str| val_str.to_string())),
+            },
+            None => Err(Error::NotFound),
+        }
+    }
```

Comment:
This would make it more readable:
```suggestion
            Some(val) => match &self.response_format {
                ResponseFormat::Imds => Mmds::format_imds_value(val),
                ResponseFormat::Json => Mmds::format_json_value(val),
            },
            None => Err(Error::NotFound),
        }
    }
    
    fn format_json_value(val: &Value) -> Result<String, Error> {
         Ok(val
                    .as_str()
                    .map_or(val.to_string(), |val_str| val_str.to_string())),
    }

```